bin/internal/update_dart_sdk.sh

```diff
@@ -60,25 +60,50 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
   }
 
   # `uname -m` may be running in Rosetta mode, instead query sysctl
-  if [ "$OS" = 'Darwin' ]; then
+  if [ -z "$ARCH" ] && [ "$OS" = 'Darwin' ]; then
     # Allow non-zero exit so we can do control flow
     set +e
-    # -n means only print value, not key
-    QUERY="sysctl -n hw.optional.arm64"
-    # Do not wrap $QUERY in double quotes, otherwise the args will be treated as
-    # part of the command
-    QUERY_RESULT=$($QUERY 2>/dev/null)
-    if [ $? -eq 1 ]; then
-      # If this command fails, we're certainly not on ARM
-      ARCH='x64'
-    elif [ "$QUERY_RESULT" = '0' ]; then
-      # If this returns 0, we are also not on ARM
-      ARCH='x64'
-    elif [ "$QUERY_RESULT" = '1' ]; then
-      ARCH='arm64'
-    else
-      >&2 echo "'$QUERY' returned unexpected output: '$QUERY_RESULT'"
-      exit 1
+    if command -v arch >/dev/null 2>&1; then
```

Comment:
Just run `ARCH_RESULT=$(arch 2>/dev/null)` in the `if` like the line below and it'll have a non-zero result if it wasn't found. Then you don't need this `if` at all.

Reply:
but we'd still need to run it 3 times, for the different paths we try.